core-web/libs/sdk/experiments/src/lib/dot-experiments.ts

```diff
@@ -99,62 +118,121 @@ export class DotExperiments {
+     * - Parses the fetched data to the form required for storage in the database.
+     * - Persists the data in the indexDB database.
      *
      * @private
+     * @method setExperimentData
      * @async
-     * @throws {Error} If an error occurs while loading the experiments.
+     * @throws {Error} Throws an error with details if there is any failure in loading the experiments or during their persistence in indexDB.
+     *
+     * @returns {Promise<void>} An empty promise that fulfills once the experiment data has been successfully loaded and persisted.
      */
     private async setExperimentData() {
         try {
-            const experimentAssignedToUser = await this.getExperimentsFromServer();
-            this.persistExperiments(experimentAssignedToUser);
+            let fetchExperiments: AssignedExperiments | null = null;
+            let storedExperiments: AssignedExperiments | null = null;
+
+            // Checks whether fetching experiment data from the server is necessary.
+            if (this.shouldCheckAnalytics()) {
+                fetchExperiments = await this.getExperimentsFromServer();
+            }
+
+            // Parses the fetched data for storage.
+            storedExperiments = this.indexDBData ? this.indexDBData.experiments : null;
+
+            const dataToPersist: AssignedExperiments = parseData(
+                fetchExperiments,
+                storedExperiments
+            );
+
+            // Persists the data in the indexDB.
+            this.persistExperiments(dataToPersist);
+
+            //
+            this.initAnalyticsClient();
         } catch (e) {
             throw Error(`Error persisting experiments to indexDB, ${e}`);
         }
     }
 
     /**
-     * Persists experiments to the IndexDB.
+     * Persists the parsed experiment data into the indexDB database.
+     *
+     * The method does the following:
+     * - Receives the parsed data.
+     * - Updates the creation date.
+     * - Clears existing data from the indexDB database.
+     * - Stores the new data in the indexDB database.
      *
-     * @param {AssignedExperiments} entity - The entity containing experiments to persist.
+     * If there are no experiments in the received data, the method will not attempt to clear or persist anything and will return immediately.
+     *
+     * Any errors encountered during storage are logged with the `dotLogger` utility.
+     *
+     * @note This method utilizes Promises for the asynchronous handling of data persistence. Errors during data persistence are caught and logged, but not re-thrown.
+     *
+     * @param {AssignedExperiments} entity - The object containing the experiment data to persist.
      * @private
+     * @method persistExperiments
+     * @throws Nothing – Errors are caught and logged, but not re-thrown.
+     *
      */
     private persistExperiments(entity: AssignedExperiments) {
+        const dataToStore = {
+            created: Date.now(),
```

Comment:
should we handle timezones here?

Reply:
No needed. This only will be used in frontend, in the client's browser. I will use this value for verify if the indexdb need a refresh.